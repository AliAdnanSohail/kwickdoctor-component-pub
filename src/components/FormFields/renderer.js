import React from 'react';
import classnames from 'classnames';

import styles, { error as errorStyles, label as labelStyles } from './styles';

export default render => ({
  input, meta, label, ...rest
}) => (
  <div className={classnames('form-field', rest.containerClassName)}>
    <label htmlFor={rest.id}>{label}</label>
    {render(input, label, {
      ...rest,
      className: classnames({
        invalid: meta && meta.error && meta.touched,
        active: meta && meta.active,
      }),
    })}
    {meta && meta.error && meta.touched ? <div className="error">{meta.error}</div> : undefined}

    <style jsx>{styles}</style>
    <style jsx>{labelStyles}</style>
    <style jsx>{errorStyles}</style>
  </div>
);
