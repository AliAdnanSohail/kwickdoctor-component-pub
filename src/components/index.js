export { default as AvatarPicker } from './AvatarPicker';
export { default as Button } from './Button';
export { default as ButtonGroup } from './ButtonGroup';
export { default as Calendar } from './Calendar';
export { default as Card } from './Card';
export { default as CardPicker } from './CardPicker';
export { default as DatePicker } from './DatePicker';
export { default as FileInput } from './FileInput';
export { default as Form } from './Form';
export { default as Modal } from './Modal';
export { default as Toast } from './Toast';
export { default as ToastManager } from './Toast/Manager';

export * from './FormFields';
export * from './Stepper';
