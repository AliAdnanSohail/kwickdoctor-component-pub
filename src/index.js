export { default as Button } from './components/Button';
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as Calendar } from './components/Calendar';
export { default as Card } from './components/Card';
export { default as Datepicker } from './components/Datepicker';
export { default as TextInput } from './components/TextInput';
export { default as Toast } from './components/Toast';
<<<<<<< HEAD
export { default as ButtonGroup } from './components/ButtonGroup';
export { default as AvatarPicker } from './components/AvatarPicker';
=======
>>>>>>> master
