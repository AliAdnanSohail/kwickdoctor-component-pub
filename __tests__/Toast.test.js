import React from 'react';
import renderer from 'react-test-renderer';

import { ToastNotification } from '../src';

const toastContentProps = {
  className: 'test-toast-content',
  text: 'test toast content',
};

function ToastContent(props) {
  return (
    <div className={props.className}>
      {props.text}
    </div>
  );
}

const initialToasts = [
  {
    id: '9c1605ae-9336-4afd-9fe4-412065ef4c9b',
    content: ToastContent(toastContentProps),
    primary: false,
    boxstyle: 'rectangle-22',
    onHideClick() {},
    timeOut: 1000,
  },
];

it('Toast renders correctly', () => {
  const element = renderer.create(<ToastNotification initialToasts={initialToasts} />).toJSON();

  expect(element).toMatchSnapshot();
});
