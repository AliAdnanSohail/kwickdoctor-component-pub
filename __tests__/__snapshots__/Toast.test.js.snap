// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toast renders correctly 1`] = `
<div
  className="toast-container"
>
  <div
    className="toast-box rectangle-22"
  >
    <div
      className="circle"
    />
    <div
      className="message"
    >
      <div
        className="test-toast-content"
      >
        test toast content
      </div>
    </div>
    <div
      className="icons-close"
    >
      <button
        className="icon"
        onClick={[Function]}
      >
        <i
          className="material-icons md-24 md-dark "
          style={
            Object {
              "color": "",
              "fontSize": "md-24",
            }
          }
        >
          close
        </i>
      </button>
    </div>
    <div
      className="clear"
    />
  </div>
  <style>
    .toast-box{max-height:100%;box-sizing:border-box;margin-bottom:10px;color:#2A2A30;-webkit-transition:-webkit-transform 200ms ease-out,opacity 200ms ease-out,max-height 100ms ease 200ms,padding 100ms ease 300ms,margin 100ms ease 400ms;-webkit-transition:transform 200ms ease-out,opacity 200ms ease-out,max-height 100ms ease 200ms,padding 100ms ease 300ms,margin 100ms ease 400ms;transition:transform 200ms ease-out,opacity 200ms ease-out,max-height 100ms ease 200ms,padding 100ms ease 300ms,margin 100ms ease 400ms;}
    .toast-box.fade-enter{opacity:0;-webkit-transform:translate(40px);-ms-transform:translate(40px);transform:translate(40px);}
    .toast-box.fade-exit-active{opacity:0;-webkit-transform:translate(40px);-ms-transform:translate(40px);transform:translate(40px);max-height:0 !important;margin:0 !important;padding:0 !important;overflow:hidden;}
    .toast-box.rectangle-22{padding:21px 18px;width:413px;border-radius:4px;background-color:#ffffff;box-shadow:0 2px 40px 0 rgba(0,0,0,0.05);}
    .toast-box .circle{margin:5px 15px 0 0;float:left;display:block;height:7px;width:7px;background-color:#222328;border-radius:50%;}
    .toast-box.primary .circle{background-color:#0c97f9;}
    .toast-box .message{float:left;width:335px;color:#222328;font-weight:500;line-height:18px;word-break:break-all;}
    .toast-box .icons-close{padding-top:2px;width:16px;float:right;}
    .toast-box .clear{clear:both;}
    .toast-box .icon{padding:0;margin:0;height:14px;width:14px;-webkit-transition:all 0.3s;transition:all 0.3s;border:none;outline:none;opacity:0.6;cursor:pointer;background-repeat:no-repeat;background-color:transparent;}
    .toast-box .icon:hover{opacity:1;}
    .toast-box .icon .material-icons{height:16px;width:16px;display:inline-block;font-size:1rem;}
    .toast-container{position:fixed;top:10px;left:50%;margin-left:-196px;}
  </style>
</div>
`;
